frontends/web/src/routes/new-settings/components/appearance/activeCurrenciesDropdownSetting.tsx

```diff
@@ -14,77 +14,20 @@
  * limitations under the License.
  */
 
-import { useEffect, useState } from 'react';
-import Select, { ActionMeta, MultiValue, MultiValueRemoveProps, components } from 'react-select';
-import { currencies, store, selectFiat, unselectFiat, SharedProps } from '../../../../components/rates/rates';
+import { store, SharedProps, currenciesWithDisplayName } from '../../../../components/rates/rates';
 import { SettingsItem } from '../settingsItem/settingsItem';
-import { Fiat } from '../../../../api/account';
 import { share } from '../../../../decorators/share';
-
-type SelectOption = {
-  label: Fiat;
-  value: Fiat;
-}
-
-type TSelectProps = {
-  options: SelectOption[];
-} & SharedProps;
-
-const ReactSelect = ({ options, active, selected }: TSelectProps) => {
-  const [selectedCurrencies, setSelectedCurrencies] = useState<SelectOption[]>([]);
-
-  useEffect(() => {
-    if (selected.length > 0) {
-      const formattedSelectedCurrencies = selected.map(currency => ({ label: currency, value: currency }));
-      setSelectedCurrencies(formattedSelectedCurrencies);
-    }
-  }, [selected]);
-
-  const MultiValueRemove = (props: MultiValueRemoveProps<SelectOption>) => {
-    const currency = props.data.value;
-    return (
-      currency !== active ?
-        <components.MultiValueRemove {...props}>
-          {'X'}
-        </components.MultiValueRemove>
-        : null
-    );
-  };
-
-  return (
-    <Select
-      classNamePrefix="react-select"
-      isSearchable
-      isClearable={false}
-      components={{ MultiValueRemove }}
-      isMulti
-      value={selectedCurrencies}
-      onChange={(selectedFiats: MultiValue<SelectOption>, meta: ActionMeta<SelectOption>) => {
-        switch (meta.action) {
-        case 'remove-value':
-          if (selectedFiats.length > 0) {
-            const unselectedFiat = meta.removedValue.value;
-            unselectFiat(unselectedFiat);
-          }
-          break;
-        case 'select-option':
-          const selectedFiat = selectedFiats[selectedFiats.length - 1].value as Fiat;
-          selectFiat(selectedFiat);
-        }
-      }}
-      options={options}
-    />);
-};
+import { ActiveCurrenciesDropdown } from '../dropdowns/activecurrenciesdropdown';
 
 const ActiveCurrenciesDropdownSetting = ({ selected, active }: SharedProps) => {
-  const formattedCurrencies = currencies.map((currency) => ({ label: currency, value: currency }));
+  const formattedCurrencies = currenciesWithDisplayName.map((fiat) => ({ label: `${fiat.displayName} (${fiat.currency})`, value: fiat.currency }));
```

Comment:
nit: this calls array map over and over but the formattedCurrencies are always the same. Maybe rates component could map once and export?

same with formattedCurrencies in defaultCurrencyDropdownSetting component (as it now always contains all currencies)

Reply:
Sure I'll do this, but these exported values are probably only gonna be used in the respective dropdown(s), as it's only used for exactly the dropdown/select component (thus it needs a `label` and `value`).